src/mongocxx/test/spec/monitoring.cpp

```diff
@@ -66,8 +67,23 @@ void apm_checker::compare_unified(bsoncxx::array::view expectations, entity::map
         return true;
     };
 
-    // This will throw an exception on unmatched fields and return true in all other cases.
-    std::equal(expectations.begin(), expectations.end(), _events.begin(), equal);
+    auto exp_it = expectations.cbegin();
+    auto exp_end = expectations.cend();
+    auto ev_it = _events.cbegin();
+    auto ev_end = _events.cend();
+    for (; exp_it != exp_end && ev_it != ev_end; ++exp_it, ++ev_it) {
+        compare(*exp_it, *ev_it);
+    }
+    if (exp_it != exp_end) {
+        auto next_expected = exp_it->get_document();
+        CAPTURE(to_json(next_expected));
+        FAIL("Not enough events occurred");
```

Comment:
Was the motivation for this change to capture the last `expectation` / `event` when the lists are not equal lengths? If so, that seems generally helpful.

Reply:
Yes. Previously it would either not check all events or it would crash.